src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs

```diff
@@ -271,6 +273,32 @@ public string GetWebAppPublishingProfile(string resourceGroupName, string webSit
             return doc.ToString();
         }
 
+        public User GetPublishingCredentials(string resourceGroupName, string webSiteName, string slotName = null)
+        {
+            string qualifiedSiteName;
+            return CmdletHelpers.ShouldUseDeploymentSlot(webSiteName, slotName, out qualifiedSiteName) ?
+               WrappedWebsitesClient.WebApps().ListPublishingCredentialsSlot(resourceGroupName, webSiteName, slotName)
+               : WrappedWebsitesClient.WebApps().ListPublishingCredentials(resourceGroupName, webSiteName);
+        }
+        public void RunWebAppContainerPSSessionScript(PSCmdlet cmdlet, string resourceGroupName, string webSiteName, string slotName = null, bool newPSSession = false)
+        {
+            Site site = GetWebApp(resourceGroupName, webSiteName, slotName);
+            User user = GetPublishingCredentials(resourceGroupName, webSiteName, slotName);
+            const string webAppContainerPSSessionVarPrefix = "webAppPSSession";
+            string publishingUserName = user.PublishingUserName.Length <= 20 ? user.PublishingUserName : user.PublishingUserName.Substring(0, 20);
+            string psSessionScript = string.Format("${3}User = '{0}' \n${3}Password = ConvertTo-SecureString -String '{1}' -AsPlainText -Force \n" +
+                "${3}Credential = New-Object -TypeName PSCredential -ArgumentList ${3}User, ${3}Password\nSet-Item WSMAN:\\LocalHost\\Client\\Auth\\Basic -Value $true \n" +
+                "Set-Item WSMAN:\\LocalHost\\Client\\TrustedHosts -Value {2} -Force\n" +
+                (newPSSession ? "${3}NewPsSession = New-PSSession" : "Enter-PSSession") + " -ConnectionUri https://{2}/WSMAN -Authentication Basic -Credential ${3}Credential \n",
+                publishingUserName, user.PublishingPassword, site.DefaultHostName, webAppContainerPSSessionVarPrefix);
+            cmdlet.ExecuteScript<object>(psSessionScript);
```

Comment:
Here is the primary issue. This code won't work on Linux (for CloudShell). Please review the email thread to see your options.

Reply:
I have removed those cmdlets from this PR for now. Will continue discussing in the email thread and eventually create a new pr when we establish a viable option. Thanks,